flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -540,6 +553,46 @@ public void testSlotReleaseAtomicallyReleasesExecution() throws Exception {
 		});
 	}
 
+	/**
+	 * Tests that producedPartitions are registered synchronously under an asynchronous interface.
+	 */
+	@Test(expected = IllegalStateException.class)
+	public void testSynchronousRegisterProducedPartitions() throws Exception {
+		ScheduledExecutorService executor = new DirectScheduledExecutorService();
```

Comment:
This is not a must, but I'd still suggest to try to mark the variables final if possible, like we do in other parts in this test class.